internal/provider/resource_access_model_test.go

```diff
@@ -147,3 +147,81 @@ resource "juju_access_model" "test" {
   users = [juju_user.test-user.name]
 }`, userName, userPassword, modelName, access)
 }
+
+func TestAcc_ResourceAccessModel_Schema_v0_To_v1(t *testing.T) {
+	SkipJAAS(t)
+	user1 := acctest.RandomWithPrefix("tfuser1")
+	password1 := acctest.RandomWithPrefix("tf-test-user1")
+	user2 := acctest.RandomWithPrefix("tfuser2")
+	password2 := acctest.RandomWithPrefix("tf-test-user2")
+	modelName := acctest.RandomWithPrefix("tf-access-model")
+
+	resourceName := "juju_access_model.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"juju": {
+						VersionConstraint: "0.21.1",
+						Source:            "juju/juju",
+					},
+				},
+				Config: testAccResourceAccessModelTwoUsers(user1, password1, user2, password2, modelName, "write", false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "model", modelName),
+					resource.TestCheckResourceAttr(resourceName, "access", "write"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "users.*", user1),
+					resource.TestCheckTypeSetElemAttr(resourceName, "users.*", user2),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: frameworkProviderFactories,
+				Config:                   testAccResourceAccessModelTwoUsers(user1, password1, user2, password2, modelName, "write", false),
+				PlanOnly:                 true,
```

Comment:
I don't think this one should have PlanOnly set.

Reply:
fixed